lumen/ai/agents.py

```diff
@@ -798,34 +756,45 @@ class BaseViewAgent(LumenBaseAgent):
 
     provides = param.List(default=["current_plot"], readonly=True)
 
+    prompt_templates = param.Dict(
+        default={
+            "main": PROMPTS_DIR / "BaseViewAgent" / "main.jinja2",
+        }
+    )
+
     async def _extract_spec(self, model: BaseModel):
         return dict(model)
 
+    @retry_llm_output()
     async def respond(
         self,
         messages: list[Message],
         render_output: bool = False,
         step_title: str | None = None
     ) -> Any:
-        pipeline = self._memory["current_pipeline"]
+        """
+        Generates a visualization based on user messages and the current data pipeline.
+        """
+        pipeline = self._memory.get("current_pipeline")
+        if not pipeline:
+            raise ValueError("No current pipeline found in memory.")
 
-        # Write prompts
-        system_prompt = await self._system_prompt_with_context(messages)
         schema = await get_schema(pipeline, include_min_max=False)
-        view_prompt = render_template(
-            "plot_agent.jinja2",
+        if not schema:
+            raise ValueError("Failed to retrieve schema for the current pipeline.")
+
+        doc = self.view_type.__doc__.split("\n\n")[0] if self.view_type.__doc__ else self.view_type.__name__
+        system_prompt = await self._render_prompt(
+            "main",
             schema=yaml.dump(schema),
             table=pipeline.table,
             current_view=self._memory.get('current_view'),
-            doc=self.view_type.__doc__.split("\n\n")[0]
+            doc=doc,
         )
-        print(f"{self.name} is being instructed that {view_prompt}.")
-
-        # Query
         output = await self.llm.invoke(
             messages,
-            system=system_prompt + view_prompt,
-            response_model=self._get_model(schema),
+            system=system_prompt,
+            response_model=self._output_type,
```

Comment:
Not correct, the output_type is the type of the object that is rendered, not the model.